File: src/main/java/com/ngolajr/ispmedia/entities/Conteudo.java
package com.ngolajr.ispmedia.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.ToString;

import java.util.UUID;

@Entity
@Data
@ToString
public class Conteudo {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    protected UUID id;
    protected String titulo;
    protected String thumbNailUri;
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    protected String descricao;
    @ManyToOne
    protected Genero genero;
    protected String editora;
    @ManyToOne
    private Utilizador criadorConteudo;
    private String dataType;
}

File: ISPMEDIA-ANGULAR/src/app/entities/Conteudo.ts
import {Genero} from "./Genero";
import {Artista} from "./Artista";
import {User} from "./User";

export class Conteudo {
  id: string=""; // UUID
  titulo: string="";
  thumbNailUri: string="";
  descricao: string="";
  genero:Genero=new Genero();
  editora:string="";
  criadorConteudo:User=new User();
  dataType:string="";

  constructor(id: string, titulo: string, thumbNailUri: string,
              descricao: string, genero:Genero,
              editora:string, criadorConteudo:User) {
    this.id = id;
    this.titulo = titulo;
    this.thumbNailUri = thumbNailUri;
    this.descricao = descricao;
    this.genero = genero;
    this.editora = editora;
    this.criadorConteudo = criadorConteudo;
  }
}
